packages/yourloops/pages/profile/profile.tsx

```diff
@@ -0,0 +1,400 @@
+
+import { Theme, createStyles, makeStyles } from "@material-ui/core/styles";
+import {
+  AppBar,
+  Breadcrumbs,
+  Button,
+  Container,
+  FormControl,
+  InputAdornment,
+  Link,
+  MenuItem,
+  Select,
+  TextField,
+  Toolbar,
+} from "@material-ui/core";
+import HomeIcon from "@material-ui/icons/Home";
+
+import HeaderBar from "../../components/header-bar";
+import { Password } from "../../components/utils/password";
+import { REGEX_BIRTHDATE, REGEX_EMAIL } from "../../lib/utils";
+import apiClient from "../../lib/auth/api";
+import { getCurrentLocaleName, getLocaleShortname, availableLocales } from "../../lib/language";
+import { Profile, Roles, Settings, Units } from "../../models/shoreline";
+
+interface Errors {
+  firstName: boolean;
+  name: boolean;
+  mail: boolean;
+  password: boolean;
+  passwordConfirmation: boolean;
+  birthDate: boolean;
+}
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    select: { padding: "1em 2em" },
+    button: { margin: "2em 1em" },
+    formControl: {
+      minWidth: 120,
+      textAlign: "right",
+    },
+    selectEmpty: {
+      marginTop: theme.spacing(2),
+    },
+    homeIcon: {
+      marginRight: "0.5em",
+    },
+    breadcrumbLink: {
+      display: "flex",
+    },
+    toolBar: {
+      display: "grid",
+      gridTemplateRows: "auto",
+      gridTemplateColumns: "auto auto auto",
+      paddingLeft: "6em",
+      paddingRight: "6em",
+    },
+    textField: {
+      "& input:disabled": {
+        backgroundColor: "#f7f7f8",
+      },
+    },
+    title: {
+      textAlign: "center",
+      color: theme.palette.primary.main,
+    },
+    inputTitle: { color: "black" },
+    inputProps: { textAlign: "right", padding: "1em 2em" },
+    container: {
+      border: "solid",
+      borderRadius: "15px",
+      marginTop: "32px",
+      // eslint-disable-next-line no-magic-numbers
+      borderColor: theme.palette.grey[300],
+      borderWidth: "1px",
+    },
+  })
+);
+
+export const ProfilePage: FunctionComponent = () => {
+  const { t, i18n } = useTranslation("yourloops");
+  const classes = useStyles();
+  const history = useHistory();
+
+  const [firstName, setFirstName] = useState<string>("");
+  const [name, setName] = useState<string>("");
+  const [mail, setMail] = useState<string>("");
+  const [locale, setLocale] = useState<string>(getCurrentLocaleName(i18n));
+  const [password, setPassword] = useState<string>("");
+  const [passwordConfirmation, setPasswordConfirmation] = useState<string>("");
+  const [unit, setUnit] = useState<Units>(Units.mole);
+  const [role, setRole] = useState<Roles | null>(null);
+  const [birthDate, setBirthDate] = useState<string>("");
+  const [hbA1c, setHbA1c] = useState<string>("8.5%"); // TODO
+  const [hasProfileChanged, setHasProfileChanged] = useState<boolean>(false);
+  const [haveSettingsChanged, setHaveSettingsChanged] = useState<boolean>(false);
+
+  useEffect(() => {
+    const user = apiClient.whoami;
```

Comment:
if possible, could you use the useAuth hook in order to get the authificated user. this whoami function will be removed.